staging/src/k8s.io/client-go/rest/client.go

```diff
@@ -78,6 +78,10 @@ type RESTClient struct {
 
 	// Set specific behavior of the client.  If not set http.DefaultClient will be used.
 	Client *http.Client
+
+	// BearerTokenSource will be used to get a token and refresh a new token, it is generated from the auth provider
+	// or the one from rest.Config
+	BearerTokenSource BearerTokenSource
```

Comment:
I still don't think a token source fits well at the top level here. Authentication plugins have been provided as a custom client before. Why do we need two ways to feed that to a rest client now?